scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+/*
+
+This test generator was initially written by David Bruce: https://github.com/davidbruce
+
+His initial PR was stale and his commits were accidentally nuked, hence this attribution.
+
+*/
+
+import haxe.Template;
+import haxe.Json;
+import sys.io.File;
+import sys.FileSystem;
+import sys.Http;
+
+using StringTools;
+
+class TestGen
+{
+	static var exercise:String;
+	static var exerciseStubTmpl = new Template("package;
+
+class ::exercise:: {
+    ::methods:: 
+}");
+
+	static var methodsTmpl = new Template("public static function ::method::(::args::): ::returnType:: {
+        throw \"Not Implemented\"; // Delete this statement and write your own implementation.
+    }");
+
+	static var testTmpl = new Template("package;
+
+using buddy.Should;
+
+//Created by testgen.hx
+class Test extends buddy.SingleSuite {
+    public function new() {
+        ::stories::
+    }
+}");
+
+	static var storyTmpl = new Template("describe(\"::story::\", {
+        ::testCases::});");
+
+	static var testCaseTmpl = new Template("    it(\"::testCase::\", {
+                ::testCriteria::
+            });
+        ");
+
+	public static function main()
+	{
+		// TODO: Configlet is able to create the directories/files we need:
+		//   ./bin/configlet sync --docs --filepaths --metadata --tests include -uye <slug>
+		// 
+		// We should try to change the code below to use configlet's
+		// cached copy of the problem-specifications repo instead
+		var dataUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/canonical-data.json");
+		var descriptionUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/description.md");
+		var metaUrl = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/metadata.toml");
+		// check if no args are passed
+		if (Sys.args().length < 1)
+		{
+			Sys.println("An exercise is required.");
+			return;
+		}
+
+		exercise = Sys.args()[0];
+
+		// pull in files from github
+		var execute = (template:Template) -> return template.execute({exercise: exercise});
+
+		var dataJson = Http.requestUrl(execute(dataUrl));
+		var descriptionMd = Http.requestUrl(execute(descriptionUrl));
+		var metaToml = Http.requestUrl(execute(metaUrl));
+
+		var exercisesDir = "../exercises/practice/";
+		var dir = '${exercisesDir}${exercise}/';
+
+		// create exercise folder
```

Comment:
Running configlet sync for the exercise will create the correct files for us so this portion where we create the missing folders / files isn't needed